cohort-engine/src/main/java/com/ibm/cohort/engine/measure/cache/TransientRetrieveCacheContext.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * (C) Copyright IBM Corp. 2020, 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.cohort.engine.measure.cache;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+
+public class TransientRetrieveCacheContext implements RetrieveCacheContext {
+
+	private Cache<CacheKey, Iterable<Object>> currentCache;
+
+	@Override
+	public Cache<CacheKey, Iterable<Object>> newCache(String patientId) {
+		this.currentCache = Caffeine.newBuilder()
+				.recordStats()
+				.maximumSize(1_000)
```

Comment:
Seems like a good place for externalized configuration. There are lots of cache tunings knobs that might need to be turned to effect best performance and resource utilization.

Reply:
This class definitely has some work to be done regarding configuration.  I just picked a reasonable default to get off the ground.  I'll add a `TODO` to be triple sure I resolve that before merging down.